docs/enhancements/image-handling-service.md

```diff
@@ -0,0 +1,155 @@
+---
+title: image-handling-service
+authors:
+  - "@carbonin"
+creation-date: 2021-07-06
+last-updated: 2021-07-07
+---
+
+# Image Handling Service
+
+## Summary
+
+Creating and serving the discovery image is a resource intensive operation which is
+significantly different from the other responsibilities of the assisted service.
+This should be handled by a separate service rather than the main application.
+
+## Motivation
+
+Most the assisted service functionality is based on many quick API calls with
+relatively small responses. Handling images is all about larger chunks of data and
+potentially longer operations. Splitting these responsibilities would allow us
+to manage and scale these deployments separately.
+
+### Goals
+
+* Create and serve discovery ISO images from a separate service
+* Make minimal changes to existing (or proposed) APIs
+
+### Non-Goals
+
+* Serve a discovery image directly to a BMC
```

Comment:
Is it worth mentioning this here as a "Non-Goal"? I only put it here because it's something that has come up in previous conversations, but I don't think there's any reason to expect that serving to a BMC would be an outcome of this proposal.

I also call this out in the "Drawbacks" section because the implementation may make this more difficult, but I think that's not the same as it being a "Non-Goal".

Reply:
@zaneb I like the implementation and I think it's something that will generally be usable in this solution.
I would like to share the embedding/streaming solution between the work we're doing, but we may need to do a bit more to make to usable in both cases. Specifically, we need to embed more than just the ignition into the iso.

Is that specific repo what you're thinking about using?

Reply:
> @zaneb I like the implementation and I think it's something that will generally be usable in this solution.
> I would like to share the embedding/streaming solution between the work we're doing, but we may need to do a bit more to make to usable in both cases. Specifically, we need to embed more than just the ignition into the iso.

Yeah, and actually we just had a discussion today about how it probably makes sense for us in the medium term to do the same thing that assisted-service is doing in serving an image sans rootfs and delivering the rest over the network.

> Is that specific repo what you're thinking about using?

I'm assuming we'll probably create a new repo for the controller/webserver and copy the code into it rather than importing it from there as an external module. But I imagine that will be the starting point.

Reply:
@zaneb why this duplication of work is required? Can you please elaborate?

Reply:
Great.  Let's do the work as described in this enhancement and then see how it can be used elsewhere.